src/screens/ExchangeModal.tsx

```diff
@@ -251,16 +303,18 @@ export default function ExchangeModal({
     updateOutputAmount,
   } = useSwapInputHandlers();
 
-  const chainId = useMemo(
-    () =>
-      ethereumUtils.getChainIdFromType(
-        inputCurrency?.type || outputCurrency?.type
-      ),
-    [inputCurrency, outputCurrency]
-  );
+  const chainId = useMemo(() => {
+    if (inputCurrency?.type || outputCurrency?.type) {
+      return ethereumUtils.getChainIdFromType(
+        inputCurrency?.type! ?? outputCurrency?.type!
```

Comment:
Youll notice im using `!` alot here, as alot of the inferred types deeper in the codebase are all potentially `undefined`, would be good to know if there is a better way of doing this in the codebase

Reply:
makes sense